docs/core/project-sdk/msbuild-props.md

```diff
@@ -524,58 +524,150 @@ The `EnableDefaultNoneItems` property controls whether `None` items (files that
+| `AllEnabledByDefault` (deprecated) | .NET 5 | Aggressive or opt-out mode, where all rules are enabled by default as build warnings. You can selectively [opt out](../../fundamentals/code-analysis/configuration-options.md) of individual rules to disable them. |
+| `AllDisabledByDefault` (deprecated) | .NET 5 | Conservative or opt-in mode, where all rules are disabled by default. You can selectively [opt into](../../fundamentals/code-analysis/configuration-options.md) individual rules to enable them. |
 
 > [!NOTE]
-> This property has no effect on code analysis in projects that don't reference a [project SDK](overview.md), for example, legacy .NET Framework projects that reference the Microsoft.CodeAnalysis.NetAnalyzers NuGet package.
+>
+> - In .NET 5 and earlier versions, this property only affects [code-quality (CAXXXX) rules](../../fundamentals/code-analysis/quality-rules/index.md). Starting in .NET 6, if you set [EnforceCodeStyleInBuild](#enforcecodestyleinbuild) to `true`, this property affects [code-style (IDEXXXX) rules](../../fundamentals/code-analysis/style-rules/index.md) too.
+> - If you use a compound value for [AnalysisLevel](#analysislevel), for example, `<AnalysisLevel>5-recommended</AnalysisLevel>`, you can omit this property entirely. However, if you specify both properties, `AnalysisLevel` takes precedence over `AnalysisMode`.
+> - This property has no effect on code analysis in projects that don't reference a [project SDK](overview.md), for example, legacy .NET Framework projects that reference the Microsoft.CodeAnalysis.NetAnalyzers NuGet package.
+
+### AnalysisMode\<Category>
+
+This property is the same as [AnalysisMode](#analysismode), except that it only applies to a specific [category of code-analysis rules](../../fundamentals/code-analysis/categories.md). This property allows you to enable or disable rules at a different level to the other rule categories. If you omit this property for a particular category of rules, it defaults to the [AnalysisMode](#analysismode) value. The available values are the same as those for [AnalysisMode](#analysismode).
+
+```xml
+<PropertyGroup>
+  <AnalysisModeSecurity>All</AnalysisModeSecurity>
+</PropertyGroup>
+```
+
+The following table lists the property name for each rule category.
+
+| Property name | Rule category |
+| - |
+| `<AnalysisModeDesign>` | Design rules |
+| `<AnalysisModeDocumentation>` | Documentation rules |
+| `<AnalysisModeGlobalization>` | Globalization rules |
+| `<AnalysisModeInteroperability>` | Portability an interoperability rules |
+| `<AnalysisModeMaintainability>` | Maintainability rules |
+| `<AnalysisModeNaming>` | Naming rules |
+| `<AnalysisModePerformance>` | Performance rules |
+| `<AnalysisModeSingleFile>` | Single-file application rules |
+| `<AnalysisModeReliability>` | Reliability rules |
+| `<AnalysisModeSecurity>` | Security rules |
+| `<AnalysisModeStyle>` | All code-style (IDEXXXX) rules |
```

Comment:
Was this designed to explicitly work for IDExxxx only? What about third-party "Style" analyzers?

Reply:
@Youssef1313 None of the categories mentioned here are reserved for first-party analyzers. Any third party analyzer can define a rule with any one of the categories mentioned above and the category based configuration will then also be applicable to that third party analyzer diagnostic. It is recommended that third party analyzers don't re-use one of these category names to allow separate bulk configuration of it's own rules from the first party rules.